packages/rsocket-examples/src/RequestStreamExample.ts

```diff
@@ -0,0 +1,44 @@
+import { RSocketConnector } from "@rsocket/rsocket-core";
+import { WebsocketClientTransport } from "@rsocket/rsocket-websocket-client";
+import { exit } from "process";
+import WebSocket from "ws";
+
+async function main() {
+  const connector = new RSocketConnector({
+    transport: new WebsocketClientTransport({
+      url: "ws://localhost:8080",
+      wsCreator: (url) => new WebSocket(url),
+    }),
+  });
+
+  const rsocket = await connector.bind();
+
+  await new Promise((resolve, reject) => {
+    const requester = rsocket.requestStream(
+      {
+        data: Buffer.from("Hello World"),
+      },
+      1,
+      {
+        onError: (e) => reject(e),
+        onNext: (payload, isComplete) => {
+          console.log(
+            `payload[data: ${payload.data}; metadata: ${payload.metadata}]|${isComplete}`
+          );
+
+          requester.request(5);
+
+          if (isComplete) {
+            resolve(payload);
+          }
```

Comment:
I'm struggling to understand in which cases `isComplete` would be true, and why `onComplete` is not sufficient for handling such a case. When would `isComplete` be true here?

Same question for `requestChannel`.

Reply:
this is a separate scenario to separate `payload(data = null, metadata= null, flags = COMPLET)` -> `onComplete()` vs `payload(data=Buffer, metadata=..., flags=NEXT|COMPLETE)` -> `onNext(payload, isComplete=true)` vs `payload(data=Buffer, metadata=..., flags=NEXT)` -> `onNext(payload, isComplete=false)`

Again, kind of introduced in Swift to reduce methods invocations. The intent here is just to keep the API solid between all the impl.

Reply:
/cc @dnadoba @nkristek

Reply:
It is an optimisation mainly for `RequestResponse`. In this case, a responder is only allowed to send one payload/onNext. Because of that, we can always set `isComplete` to true to reduce the number of frames we send from two (on with the payload and one with just the isComplete flag set) to just one (with the payload and the isComplete flag set). 

This optimisation could also be used for the last payload/onNext of a RequestStream/RequestChannel but we currently can not detect that in an easy way automatically in Swift without delaying/buffering frames.

Reply:
Thanks for the explanations @OlegDokuka @dnadoba . Those optimization scenarios make sense from the perspective of the protocol and packet transmission, however, I'm wondering if we can offer a bit of a more ergonomic API from the perspective of the receiving stream.

I'm envisioning that rather than pass the `isComplete` flag to `onNext`, the stream handler could instead invoke `onNext(payload)` followed by `onComplete()`.

```ts
this.receiver.onNext(payload);
if (hasCompleteFlag) {
  this.receiver.onComplete();
}
```

For the time being, we won't let these questions block this PR, however, let's try and find some time to discuss as a group, as like @OlegDokuka mentioned, we'll be looking to base future Implementations off of the president that RSocket-Switft and now RSocket-JS are setting.